pkg/controller/plan/migration.go

```diff
@@ -1481,52 +1481,54 @@ func (r *Migration) updateConversionProgress(vm *plan.VMStatus, step *plan.Step)
 		return
 	}
 
-	if pod != nil {
-		switch pod.Status.Phase {
-		case core.PodSucceeded:
-			step.MarkCompleted()
-			step.Progress.Completed = step.Progress.Total
-		case core.PodFailed:
-			step.MarkCompleted()
-			step.AddError("Guest conversion failed. See pod logs for details.")
-		default:
-			el9, el9Err := r.Context.Plan.VSphereUsesEl9VirtV2v()
-			if el9Err != nil {
-				err = el9Err
-				return
-			}
-			if el9 {
-				err = r.updateConversionProgressEl9(pod, step)
-				if err != nil {
-					// Just log it. Missing progress is not fatal.
-					log.Error(err, "Failed to update conversion progress")
-					err = nil
-					return
-				}
-			}
-		}
-	} else {
+	if pod == nil {
 		step.MarkCompleted()
 		step.AddError("Guest conversion pod not found")
+		return
+	}
+
+	switch pod.Status.Phase {
+	case core.PodSucceeded:
+		step.MarkCompleted()
+		step.Progress.Completed = step.Progress.Total
+	case core.PodFailed:
+		step.MarkCompleted()
+		step.AddError("Guest conversion failed. See pod logs for details.")
+	default:
+		if pod.Status.PodIP == "" {
+			// we get the progress from the pod and we cannot connect to the pod without PodIP
+			break
+		}
+
+		el9, el9Err := r.Context.Plan.VSphereUsesEl9VirtV2v()
+		switch {
+		case el9Err == nil && el9:
+			if progressErr := r.updateConversionProgressEl9(pod, step); progressErr != nil {
+				// Just log it. Missing progress is not fatal.
+				log.Error(err, "Failed to update conversion progress")
+			}
+		case el9Err != nil:
+			err = el9Err
+			return
+		}
 	}
+
 	return
 }
 
 func (r *Migration) updateConversionProgressEl9(pod *core.Pod, step *plan.Step) (err error) {
-	if pod.Status.PodIP == "" {
```

Comment:
but then you build a bad url which you later do nothing with it(trying to get) and then you will get an error?

Reply:
i did not.